src/intl-head.tsx

```diff
@@ -0,0 +1,23 @@
+import Head from 'next/head';
+import { useRouter } from 'next/router';
+import React, { ReactElement, ReactNode } from 'react';
+import { useAlternateLinks } from './hooks/useAlternateLinks';
+
+interface IntlHeadProps {
+  children?: ReactNode;
+  currentLocale?: string;
+}
+
+export function IntlHead({ children, currentLocale }: IntlHeadProps): ReactElement {
+  const { locale } = useRouter();
+  const alternateLinks = useAlternateLinks(currentLocale ?? locale);
```

Comment:
This fallback should be happening inside `useAlternateLinks`, rather than outside it. That way its `locale` argument can be optional, defaulting to whatever the router considers as the current locale.

Reply:
I've actually realized that I don't need to pass the locale, I wasn't being used in `useAlternateLinks`